ingest/expression_files/mtx.py

```diff
@@ -183,64 +194,96 @@ def transform(self):
                 if not is_sorted:
                     raise ValueError("MTX file must be sorted")
                 else:
+
                     visited_expression_idx.append(current_idx)
-                    # Create data arrays from prior gene
+                    # Add current gene's gene model
+                    # Data array for cell names
                     if last_idx != 0:
-                        # Data array for cell names
-                        da_cells = GeneExpression.create_data_arrays(
-                            name=gene,
-                            array_type=f"{gene} Cells",
+                        last_gene_id, last_gene = self.genes[current_idx - 2].split(
+                            "\t"
+                        )
```

Comment:
I've been reading this code for the past 5 minutes and can't figure it out.  Let's find a time to chat--I assume it's doing something very smart that I just can't figure out.